ts/screens/onboarding/OnboardingServicesPreferenceScreen.tsx

```diff
@@ -1,179 +1,214 @@
-    useManualConfigBottomSheet(() =>
-      props.onServicePreferenceSelected(ServicesPreferencesModeEnum.MANUAL)
-    );
+    useManualConfigBottomSheet(() => {
+      selectCurrentMode(ServicesPreferencesModeEnum.MANUAL);
+    });
 
   const handleOnSelectMode = (mode: ServicesPreferencesModeEnum) => {
     // if user's choice is 'manual', open bottom sheet to ask confirmation
     if (mode === ServicesPreferencesModeEnum.MANUAL) {
       confirmManualConfig();
       return;
     }
-    setModeSelected(mode);
+    selectCurrentMode(mode);
   };
 
+  useOnFirstRender(() => {
+    trackServiceConfigurationScreen(getFlowType(true, isFirstOnboarding));
+  });
+
+  useEffect(() => {
+    // show error toast only when the profile updating fails
+    // otherwise, if the profile is in error state,
+    // the toast will be shown immediately without any updates
+    if (
+      prevProfile !== undefined &&
+      !pot.isError(prevProfile) &&
+      pot.isError(profile)
+    ) {
+      IOToast.error(I18n.t("global.genericError"));
+      return;
+    }
+
+    // if profile preferences are updated correctly
+    // the button is selected
+    // and the success banner is shown
+    if (
+      prevProfile !== undefined &&
+      pot.isUpdating(prevProfile) &&
+      pot.isSome(profile)
+    ) {
+      setModeSelected(profileServicePreferenceMode);
+      IOToast.success(
+        profileServicePreferenceMode === ServicesPreferencesModeEnum.MANUAL
+          ? I18n.t("services.optIn.preferences.manualConfig.successAlert")
+          : I18n.t("services.optIn.preferences.quickConfig.successAlert")
+      );
+    }
+  }, [prevProfile, profile, profileServicePreferenceMode]);
+
   // show a badge when the user is not new
+  // As explained in this comment (https://pagopa.atlassian.net/browse/IOPID-1511?focusedCommentId=126354)
+  // a future feature will be need this value,
+  // so I didn't delete it even though it is not used
   const showBadge = !isFirstOnboarding;
   return (
-    <RNavScreenWithLargeHeader
-      title={{
-        label: I18n.t("services.optIn.preferences.title")
-      }}
-      description={I18n.t("services.optIn.preferences.body")}
-      headerActionsProp={{ showHelp: true }}
-      contextualHelp={emptyContextualHelp}
-      fixedBottomSlot={
-        <SafeAreaView>
-          <FooterWithButtons
-            type={"SingleButton"}
-            leftButton={{
-              ...confirmButtonProps(handleOnContinue),
-              disabled: !isServicesPreferenceModeSet(modeSelected)
-            }}
-          />
+    <LoadingSpinnerOverlay isLoading={isLoading}>
+      <RNavScreenWithLargeHeader
+        title={{
+          label: I18n.t("services.optIn.preferences.title")
+        }}
+        description={I18n.t("services.optIn.preferences.body")}
+        headerActionsProp={{ showHelp: true }}
+        contextualHelp={emptyContextualHelp}
+        fixedBottomSlot={
+          <SafeAreaView>
+            <FooterWithButtons
```

Comment:
Does the new `FooterWithButtons` already handle the `SafeAreaInsets`?

Reply:
done!